src/common/cp_para_types.F

```diff
@@ -42,15 +39,18 @@ MODULE cp_para_types
 !>      08.2002 created [fawzi]
 !> \author Fawzi Mohamed
 ! **************************************************************************************************
-   TYPE cp_para_env_type
-      LOGICAL :: owns_group, ionode
-      INTEGER :: mepos, source, num_pe, ref_count
-      TYPE(mp_comm_type) :: group = mp_comm_null
+   TYPE, EXTENDS(mp_comm_type) :: cp_para_env_type
```

Comment:
Is there a reason we have to keep the `cp_para_env_type`?
Couldn't we just add those few fields to the `mp_comm_type` and then use it everywhere?

Reply:
The cp_para_env_type implements reference counting and the owns_group variable to consider shared ownership, the mp_comm_type doesn't. The cp_para_env_type is meant to be used for pointers whereas the mp_comm_type is employed for local operations. Then, reference counting is not necessary.
In this spirit, I will keep the cp_para_env module which implements routines to deal with POINTERs including some sanity checks. Because they rely on CPABORT, I cannot merge them with cp_para_types due to potential circular dependencies.
Nevertheless, I will shift the meta-data to the mp_comm_type. This allows some adjustments to calls to the MPI_Bcast because their source process is often equivalent to the source component of the para_env.